### ISSUE TITLE
[2.5.x] Impossible to process requests having a gzipped body

### ISSUE THREAD

**Issue by nremond** [playframework/playframework#6152]
### Play Version (2.5.x / etc)
2.5.3 (regression in the 2.5.x branch as it is working on the 2.4.x one)

### API (Scala / Java / Neither / Both)
Both

### Operating System (Ubuntu 15.10 / MacOS 10.10 / Windows 10)

Ubuntu 14.04 / MacOS 10.10

### JDK (Oracle 1.8.0_72, OpenJDK 1.8.x, Azul Zing)

1.8.0_65

### Library Dependencies

Vanilla Play setup

### Expected Behavior

Please describe the expected behavior of the issue, starting from the first action.

1. Send a request with a gzipped json body

### Actual Behavior

Please provide a description of what actually happens, working from the same starting point.

1.  `For request 'POST /foo' [Invalid Json: No content to map due to end-of-input
 at [Source: akka.util.ByteIterator$ByteArrayIterator$$anon$1@168de375; line: 1, column: 0]]` - it seems that the body is NOT unzipped
2. By truning `play.server.netty.log.wire=true`, it seems weird that the `Content-Encoding` is not present in the request reiceved from Netty

### Reproducible Test Case

Here is a very minimal project, see the README for the details on how to reproduce:
https://github.com/nremond/play25-gzip-request/tree/master

**Comment by wsargent** [comment]
In your example, you're using 

```
$curl -H "Content-Type: application/json" -H "Content-Encoding: gzip" -X POST http://localhost:9000/foo --data-binary "@body.json.gz"
```

Please try with `Transfer-Encoding`:

```
curl -H "Content-Type: application/json" -H "Transfer-Encoding: gzip" -X POST http://localhost:9000/foo --data-binary "@body.json.gz"
```

**Comment by wslulciuc** [comment]
@wsargent I'm also seeing this behavior.

**uncompressed**:

**Comment by schmitch** [comment]
actually the content type of gzip uploaded body is not json.

**Comment by schmitch** [comment]
@wsargent this is a bug. only happens on bodies with a Content-Lengt if you upload via -H 'Transfer-Encoding: chunked' -H 'Content-Encoding: gzip' -H 'Content-Type: application/json' it will work.

Just for reference:
```
schmitch@SHANGHAI:~$ curl -i --compressed -v -H 'Content-Type: application/json' -X POST http://localhost:9090 -H 'Content-Encoding: gzip' --header "Transfer-Encoding: chunked" --data-binary @demo.json.gz
* Rebuilt URL to: http://localhost:9090/
*   Trying ::1...
* Connected to localhost (::1) port 9090 (#0)
> POST / HTTP/1.1
> Host: localhost:9090
> User-Agent: curl/7.43.0
> Accept: */*
> Accept-Encoding: deflate, gzip
> Content-Type: application/json
> Content-Encoding: gzip
> Transfer-Encoding: chunked
> 
> 30
* upload completely sent off: 55 out of 48 bytes
< HTTP/1.1 200 OK
HTTP/1.1 200 OK
< Content-Length: 0
Content-Length: 0
< Date: Fri, 17 Jun 2016 22:50:50 GMT
Date: Fri, 17 Jun 2016 22:50:50 GMT

< 
* Connection #0 to host localhost left intact
schmitch@SHANGHAI:~$ curl -i --compressed -v -H 'Content-Type: application/json' -X POST http://localhost:9090 -H 'Content-Encoding: gzip' --data-binary @demo.json.gz
* Rebuilt URL to: http://localhost:9090/
*   Trying ::1...
* Connected to localhost (::1) port 9090 (#0)
> POST / HTTP/1.1
> Host: localhost:9090
> User-Agent: curl/7.43.0
> Accept: */*
> Accept-Encoding: deflate, gzip
> Content-Type: application/json
> Content-Encoding: gzip
> Content-Length: 48
> 
* upload completely sent off: 48 out of 48 bytes
< HTTP/1.1 400 Bad Request
HTTP/1.1 400 Bad Request
< Content-Length: 2256
Content-Length: 2256
< Content-Type: text/html; charset=utf-8
Content-Type: text/html; charset=utf-8
< Date: Fri, 17 Jun 2016 22:50:59 GMT
Date: Fri, 17 Jun 2016 22:50:59 GMT

< 


<!DOCTYPE html>
<html lang="en">
    <head>
        <title>Bad Request</title>
        <link rel="shortcut icon" href="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAAlFJREFUeNqUU8tOFEEUPVVdNV3dPe8xYRBnjGhmBgKjKzCIiQvBoIaNbly5Z+PSv3Aj7DSiP2B0rwkLGVdGgxITSCRIJGSMEQWZR3eVt5sEFBgTb/dN1yvnnHtPNTPG4PqdHgCMXnPRSZrpSuH8vUJu4DE4rYHDGAZDX62BZttHqTiIayM3gGiXQsgYLEvATaqxU+dy1U13YXapXptpNHY8iwn8KyIAzm1KBdtRZWErpI5lEWTXp5Z/vHpZ3/wyKKwYGGOdAYwR0EZwoezTYApBEIObyELl/aE1/83cp40Pt5mxqCKrE4Ck+mVWKKcI5tA8BLEhRBKJLjez6a7MLq7XZtp+yyOawwCBtkiBVZDKzRk4NN7NQBMYPHiZDFhXY+p9ff7F961vVcnl4R5I2ykJ5XFN7Ab7Gc61VoipNBKF+PDyztu5lfrSLT/wIwCxq0CAGtXHZTzqR2jtwQiXONma6hHpj9sLT7YaPxfTXuZdBGA02Wi7FS48YiTfj+i2NhqtdhP5RC8mh2/Op7y0v6eAcWVLFT8D7kWX5S9mepp+C450MV6aWL1cGnvkxbwHtLW2B9AOkLeUd9KEDuh9fl/7CEj7YH5g+3r/lWfF9In7tPz6T4IIwBJOr1SJyIGQMZQbsh5P9uBq5VJtqHh2mo49pdw5WFoEwKWqWHacaWOjQXWGcifKo6vj5RGS6zykI587XeUIQDqJSmAp+lE4qt19W5P9o8+Lma5DcjsC8JiT607lMVkdqQ0Vyh3lHhmh52tfNy78ajXv0rgYzv8nfwswANuk+7sD/Q0aAAAAAElFTkSuQmCC">
        <style>
            html, body, pre {
                margin: 0;
                padding: 0;
                font-family: Monaco, 'Lucida Console', monospace;
                background: #ECECEC;
            }
            h1 {
                margin: 0;
                background: #AD632A;
                padding: 20px 45px;
                color: #fff;
                text-shadow: 1px 1px 1px rgba(0,0,0,.3);
                border-bottom: 1px solid #9F5805;
                font-size: 28px;
            }
            p#detail {
                margin: 0;
                padding: 15px 45px;
                background: #F6A960;
                border-top: 4px solid #D29052;
                color: #733512;
                text-shadow: 1px 1px 1px rgba(255,255,255,.3);
                font-size: 14px;
                border-bottom: 1px solid #BA7F5B;
            }
        </style>
    </head>
    <body>
        <h1>Bad Request</h1>

        <p id="detail">
            For request 'POST /' [Invalid Json: No content to map due to end-of-input
 at [Source: akka.util.ByteIterator$ByteArrayIterator$$anon$1@7c9c4ef4; line: 1, column: 0]]
        </p>

    </body>
</html>







* Connection #0 to host localhost left intact
```

**Comment by gmethvin** [comment]
This is a netty bug: netty/netty#5428. It will be fixed in the next release of netty, and #6257 is a workaround.

**Comment by nremond** [comment]
Thank you @schmitch & @gmethvin !
Where would it be a good place to have a regression test for this?
I created that external project because I didn't really know where to go to write a test.
